### PR TITLE
Added info.json for mt980 keyboard and fixes to walker keymap

### PR DESCRIPTION
**Pull request by walkerstop**
Added info.json for mt980 keyboard so configurator works on this keyboard, and fix a couple issues in keymap

### CODE REVIEW

File: keyboards/mt980/info.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        {"label":"K14", "x":4.25, "y":4.5},
+        {"label":"K15", "x":5.25, "y":4.5},
+        {"label":"K16", "x":6.25, "y":4.5},
+        {"label":"K17", "x":7.25, "y":4.5},
+        {"label":"K18", "x":8.25, "y":4.5},
+        {"label":"K78", "x":9.25, "y":4.5},
+        {"label":"K77", "x":10.25, "y":4.5},
+        {"label":"K75", "x":11.25, "y":4.5},
+        {"label":"K74", "x":12.25, "y":4.5, "w":1.75},
+        {"label":"K76", "x":15.5, "y":4.5},
```
Comment (noroadsleft) [inline keyboards/mt980/info.json]: K73 should be before K76.

```suggestion
        {"label":"K73", "x":14.25, "y":4.75},
        {"label":"K76", "x":15.5, "y":4.5},
```

File: keyboards/mt980/info.json [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        {"label":"K72", "x":16.5, "y":4.5},
+        {"label":"K71", "x":17.5, "y":4.5},
+        {"label":"K70", "x":18.5, "y":4.5},
+        {"label":"K73", "x":14.25, "y":4.75},
```
Comment (noroadsleft) [inline keyboards/mt980/info.json]:
```suggestion
```

File: keyboards/mt980/info.json [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        {"label":"K00", "x":0, "y":5.5, "w":1.25},
+        {"label":"K01", "x":1.25, "y":5.5},
+        {"label":"K02", "x":2.25, "y":5.5, "w":1.25},
+        {"label":"K06", "x":3.5, "y":5.5, "w":6.25},
+        {"label":"K08", "x":9.75, "y":5.5},
+        {"label":"K07", "x":10.75, "y":5.5},
+        {"label":"K05", "x":11.75, "y":5.5, "w":1.25},
+        {"label":"K62", "x":16.5, "y":5.5},
```
Comment (noroadsleft) [inline keyboards/mt980/info.json]: K04, K03 and K66 should be before K62.

```suggestion
        {"label":"K04", "x":13.25, "y":5.75},
        {"label":"K03", "x":14.25, "y":5.75},
        {"label":"K66", "x":15.25, "y":5.75},
        {"label":"K62", "x":16.5, "y":5.5},
```

File: keyboards/mt980/info.json [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        {"label":"K61", "x":17.5, "y":5.5},
+        {"label":"K60", "x":18.5, "y":5.5},
```
Comment (noroadsleft) [inline keyboards/mt980/info.json]:
```suggestion
        {"label":"K60", "x":18.5, "y":5.5}
```

File: keyboards/mt980/info.json [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        {"label":"K04", "x":13.25, "y":5.75},
```
Comment (noroadsleft) [inline keyboards/mt980/info.json]:
```suggestion
```

File: keyboards/mt980/info.json [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        {"label":"K03", "x":14.25, "y":5.75},
```
Comment (noroadsleft) [inline keyboards/mt980/info.json]:
```suggestion
```

File: keyboards/mt980/info.json [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        {"label":"K66", "x":15.25, "y":5.75}
```
Comment (noroadsleft) [inline keyboards/mt980/info.json]:
```suggestion
```

**Comment by mechmerlin** [comment]
That's very odd. @noroadsleft I gave @walkerstop your tool to convert kbfirmware.json file over. Walker, did you use it? If you did, we might have to look at that tool again.

**Comment by walkerstop** [comment]
> That's very odd. @noroadsleft I gave @walkerstop your tool to convert kbfirmware.json file over. Walker, did you use it? If you did, we might have to look at that tool again.

Yep I used that tool

File: keyboards/mt980/mt980.c [hunk 8]
```diff
@@ -20,6 +20,18 @@ void led_set_keymap(uint8_t usb_led) {}
...
 
+void matrix_init_user(void) {
+  matrix_init_keymap();
+}
+
+void matrix_scan_user(void) {
+  matrix_scan_keymap();
+}
+
+void action_function_user(keyrecord_t *record, uint8_t id, uint8_t opt) {
```
Comment (drashna) [inline keyboards/mt980/mt980.c]: This doesn't actually exist. 

Also, functions are depreciated, and should be removed.

**Comment by noroadsleft** [comment]
> > That's very odd. @noroadsleft I gave @walkerstop your tool to convert kbfirmware.json file over. Walker, did you use it? If you did, we might have to look at that tool again.
> 
> Yep I used that tool

@walkerstop, any chance you could send me the JSON you put into the tool? I could use it for debugging.

I think I already know what this issue is, but I'd need an appropriate JSON file to solve it.

File: keyboards/mt980/mt980.c (hunk 8 above)
Comment (walkerstop) [reply keyboards/mt980/mt980.c]: OK removed it

**Comment by walkerstop** [comment]
> > > That's very odd. @noroadsleft I gave @walkerstop your tool to convert kbfirmware.json file over. Walker, did you use it? If you did, we might have to look at that tool again.
> > 
> > 
> > Yep I used that tool
> 
> @walkerstop, any chance you could send me the JSON you put into the tool? I could use it for debugging.
> 
> I think I already know what this issue is, but I'd need an appropriate JSON file to solve it.

Sure, here it is https://drive.google.com/open?id=1vfHCIwZ5Uaao3eeqXfxaPmrOfW9vGRO7

**Comment by noroadsleft** [comment]
@walkerstop @mechmerlin 

K, I have a partial answer for why the `info.json` was out of order.

The way my tool parses/generates the code right now, every unique `y` value is on a different line of the generated code, so in this case, the argument in the layout macro for the Up arrow isn't on the same line as the Shift keys:

```
#define LAYOUT( \
  K50, K52, K53, K54, K55, K57, K58, KB8, KB7, KB5, KB4, KB3, KB6, K51, KB2, KB1, KB0, K63, \
  K40, K41, K42, K43, K44, K45, K46, K47, K48, KA8, KA7, KA5, KA4, KA3, KA6, KA2, KA1, KA0, K64, \
  K30, K31, K32, K33, K34, K35, K36, K37, K38, K98, K97, K95, K94, K84, K96, K92, K91, K90, \
  K20, K21, K22, K23, K24, K25, K26, K27, K28, K88, K87, K85, K93, K86, K82, K81, K80, \
  K10, K11, K12, K13, K14, K15, K16, K17, K18, K78, K77, K75, K74, K76, K72, K71, K70, \
  K73, \
  K00, K01, K02, K06, K08, K07, K05, K62, K61, K60, \
  K04, K03, K66  \
)
```

K73 is the Up arrow key, and K04, K03 and K66 are the Left, Down and Right arrows. (K74 is Right Shift, and K76 is KC_P1.)

From that code generation (ran walkerstop's linked file through the tool), the `info.json` output is valid, but only if the above code block is the physical layout macro.

walkerstop didn't use the generated `<keyboard>.h` output though, but used an output that was more in line with current convention (which is one of the outstanding issues with output from the KBFirmware-to-QMK tool). The `info.json` output, however, *was used*, and was not updated to match, which is where the mismatch happened. A "symptom" of this is that the `label` keys in `info.json` here don't match the argument identifiers in the existing `mt980.h` (my tool uses the same identifiers between these files; the `label` in the JSON is the same as the argument identifier in `<keyboard>.h`).

I do intend to solve this, but I've been stumped on this front for a while. My goal is that a user can dump a KBFirmware-format JSON file in, and get usable QMK code out that requires nothing more than saving the appropriate output to the appropriate file. I still at this point consider the tool to be very much in Beta.

**Comment by mechmerlin** [comment]
This is good to know for future ports, thanks @noroadsleft

**Pull request merged**